pkg/workflow/providers/oam/apply.go

```diff
@@ -146,10 +206,82 @@ func lookUpValues(v *value.Value) (*common.ApplicationComponent, *value.Value, s
 	return comp, patcher, clusterName, overrideNamespace, env, nil
 }
 
+func (p *provider) loadDynamicComponent(comp *common.ApplicationComponent) (*common.ApplicationComponent, error) {
+	if comp.Type != "ref-objects" {
+		return comp, nil
+	}
+	_comp := comp.DeepCopy()
+	props := &struct {
+		Objects []struct {
+			APIVersion string            `json:"apiVersion"`
+			Kind       string            `json:"kind"`
```

Comment:
`Kind` is not the proper resource identifier, it's used for serialize/deserialize general kubernetes resources. we dont involve codec if we're merely referencing them. in addition to that, the version is neglectable, b/c api-discovery will suggest the preferred version. we should use `group` + `resource`(i.e. plural lowercased form) + `name` + `namespace` (optional) to identify the resources